### PR TITLE
Add login with AAD support

### PR DESCRIPTION
**Pull request by sophiaso**

### CODE REVIEW

File: initializr-web/src/main/resources/static/js/auth.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+(function () {
+
+    // Enter Global Config Values & Instantiate ADAL AuthenticationContext
+    window.config = {
+        instance: 'https://login.partner.microsoftonline.cn/',
```
Comment (yaweiw) [inline initializr-web/src/main/resources/static/js/auth.js:5]: After having a determined FQDN, we will need to register a multi-tenant app to allow anyone sign in. 
https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-devhowto-multi-tenant-overview

File: initializr-web/src/main/resources/templates/home.mustache [hunk 2]
```diff
@@ -18,23 +18,30 @@ new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
...
                 </a>
             </div>
-            <div class="navbar-menu">
+            <div class="navbar-menu is-active">
                 <div class="navbar-start">
                     <h1 class="navbar-item title has-text-white" >
                         Azure Spring Cloud Playground
                     </h1>
                 </div>
+                <div class="navbar-end" id="login_link">
```
Comment (yaweiw) [inline initializr-web/src/main/resources/templates/home.mustache]: Can we also add sth. like below?
```
<a href="https://azure.microsoft.com/en-us/free/" id="mobile-navigation-free-account" class="header-button-primary" title="Free account" data-event="global-clicked-primarycta">Free account</a>
```

Link: https://azure.microsoft.com/en-us/

Comment (sophiaso) [reply initializr-web/src/main/resources/templates/home.mustache]: OK

File: initializr-web/src/main/resources/static/js/auth.js (hunk 1 above)
Comment (sophiaso) [reply initializr-web/src/main/resources/static/js/auth.js:5]: Create an issue to track this? Or we do this now?

File: initializr-web/src/main/resources/templates/home.mustache (hunk 2 above)
Comment (sophiaso) [reply initializr-web/src/main/resources/templates/home.mustache]: Free account link added.

**Review by yaweiw — APPROVED** [review]

**Comment by sophiaso** [comment]
refs #40

**Pull request merged**